### DISCUSSION TITLE
Course load discussion

### DISCUSSION THREAD

**Discussion by darshan-bhojani** [snigi-gupta/AttendingUniversityAtBuffalo#3] [Q&A] [answered]
Hi Snigdha, Darshan here. So I have few questions about the courses that I enrolled in, it would be great if you could help me out. These are the courses I am currently enrolled in:-
586 Distributed Systems - Ethan Blanton
562 Database Systems - Oliver Kennedy
574 Introduction to Machine Learning - Sreyasse Das Bhattacharjee
531 Analysis of Algorithms - Shi Li

Following are the questions I have:-
1. How heavy will the course load be considering the above courses and professors? Is it manageable if I do on-campus job?
2. And it would be great if you can suggest some concepts, libraries, languages, etc that I can brush up on.
3. Also if the course work would be heavy could you recommend other courses for Software specialization? I have these in mind - 560 - Data Models and Query Languages(Sreyasse Das Bhattacharjee), 542 Software Engineering Concepts(Michael F. Buckley), 505 - Fundamentals of Programming Languages(Bharat Jayaraman).

Again thanks for hosting such informative webinars, they have helped us a lot and made the process a lot easier for everyone.

**Comment by snigi-gupta** [comment, marked as answer]
@darshan-bhojani,

DS and DB are both great courses. Definitely heavy courses so you should be very confident of taking both of them together in one sem. 
Algorithms is also medium-heavy. It all depends on how well you are in DSA (Data structures and Algorithms) already. If you do know basic algorithms like sorting (merge, quick, heap), Dynamic programming (at least have a basic understanding), Divide and conquer (basic understanding) then Algorithms should not be very difficult for you. Additionally, you should also be a decent leetcoder (able to solve Easy Leetcode questions)

ML by Das is something I did not pursue so I don't have thought about that. You can audit the course and check her rating on rate my professor.

I would also suggest you read Shi Li's course review on my Github. It might give you some perspective.

All in all, if your algos is not strong then this workload can be pretty heavy. If your algos is smooth, then you can manage a decent workload.  Alternatively, you can take algos in next semester and take an easier elective or seminar to ensure you take both DS and DB which infact are great choices.

**Reply by darshan-bhojani** [reply]
Thank you for the quick reply, this definitely cleared my doubts.

**Reply by snigi-gupta** [reply]
@darshan-bhojani ,

I also replied to other parts of your questions. Sorry I missed it earlier.

**Comment by snigi-gupta** [comment]
@darshan-bhojani,

I realized you had multiple questions and I only answered a part of them. 

1. How heavy will the course load be considering the above courses and professors? Is it manageable if I do on-campus job?
- As I said the course is pretty heavy. I will certainly not suggest to do part-time job with this workload. If you are too confident you will manage then only go for it. CDS jobs can be pretty tiring since they have manual labor involved.

2. And it would be great if you can suggest some concepts, libraries, languages, etc that I can brush up on.
- Brush up Python, DSA, for ML I have mentioned some reading (refer #2 ) and just make sure you read the material/ppt in advance of the class to stay in sync with lectures.

3. Also if the course work would be heavy could you recommend other courses for Software specialization? I have these in mind - 560 - Data Models and Query Languages(Sreyasse Das Bhattacharjee), 542 Software Engineering Concepts(Michael F. Buckley), 505 - Fundamentals of Programming Languages(Bharat Jayaraman).
- I haven't taken any of these courses by these professors. I took DMQL by Cheung. I did not like the way course was structured. I am not sure how Das's DMQL will be. This course is about SQL, ER diagrams, recursive SQL, joins, XML etc.  SEC is also something I have not pursued but Buckly and Jayraman are both good professors.
